src/main/scala/com/scalacamp/hometasks/web/ops/UserJsonProtocol.scala

```diff
@@ -0,0 +1,35 @@
+package com.scalacamp.hometasks.web.ops
+
+import com.scalacamp.hometasks.web.domain.User
+import spray.json._
+
+object UserJsonProtocol extends DefaultJsonProtocol {
+
+  implicit object UserJsonFormat extends RootJsonFormat[User] {
+
+    override def read(json: JsValue): User = {
+      json.asJsObject.getFields("username", "address", "email") match {
+        case Seq(JsString(username), JsString(address), JsString(email)) => User(0, username, Some(address), email)
+        case Seq(JsString(username), JsString(email)) => User(0, username, None, email)
+        case _ => throw DeserializationException("User expected")
+      }
+    }
+
+    override def write(obj: User): JsValue = {
+      obj.address match {
+        case Some(address) => JsObject(
+          "id" -> obj.id.toJson,
+          "username" -> obj.username.toJson,
+          "address" -> address.toJson,
+          "email" -> obj.email.toJson
+        )
+        case None => JsObject(
+          "id" -> obj.id.toJson,
+          "username" -> obj.username.toJson,
+          "email" -> obj.email.toJson
+        )
+      }
```

Comment:
all this could be simplified to smth like:
```scala
import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
import spray.json.DefaultJsonProtocol

trait JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
  implicit val userFormat = jsonFormat4(User.apply)
}
```

Reply:
Actually, I tried this at first, but in this case in user post request id field should be specified

Reply:
ideally you should use another case class for post, like case class RegisterUserRequest() without id